src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -243,12 +243,30 @@ public GetMapRequest read(Object request, Map kvp, Map rawKvp) throws Exception
         // set the raw params used to create the request
         getMap.setRawKvp(rawKvp);
 
+        boolean citeCompliant = wms.getServiceInfo().isCiteCompliant();
+
         // wms 1.3, srs changed to crs
         if (kvp.containsKey("crs")) {
             getMap.setSRS((String) kvp.get("crs"));
+        } else if (citeCompliant && WMS.VERSION_1_3_0.equals(WMS.version(getMap.getVersion()))) {
+            throw new ServiceException(
+                    "GetMap CRS parameter is missing, it is mandatory for CITE compliant mode");
         }
         // do some additional checks
 
+        if (citeCompliant && rawKvp != null && rawKvp.containsKey("transparent")) {
+            String trans = (String) rawKvp.get("transparent");
+
+            if (!trans.equalsIgnoreCase("false")
+                    && !trans.equalsIgnoreCase("true")
+                    && !trans.equalsIgnoreCase("0")
```

Comment:
See above, "transparent" does not seem to allow for "0" or "1", where are they coming from?

Quoting from the specification:

> The optional TRANSPARENT parameter specifies whether the map background is to be made transparent or not.
> TRANSPARENT can take on two values, "TRUE" or "FALSE". The default value is FALSE if this parameter is
> absent from the request.

Reply:
Thank you very much @aaime for your review and support.